tests/beacon/test_deposit_helpers.py

```diff
@@ -0,0 +1,221 @@
+@pytest.mark.parametrize(
+    "balance,"
+    "latest_status_change_slot,"
+    "zero_balance_validator_ttl,"
+    "current_slot,"
+    "expected",
+    (
+        (0, 1, 1, 2, 0),
+        (1, 1, 1, 2, None),  # not (balance == 0)
+        (0, 1, 1, 1, None),  # not (validator.latest_status_change_slot + zero_balance_validator_ttl <= current_slot) # noqa: E501
+    ),
+)
+def test_min_empty_validator_index(sample_validator_record_params,
+                                   balance,
+                                   latest_status_change_slot,
+                                   zero_balance_validator_ttl,
+                                   current_slot,
+                                   expected):
+    validators = [
+        ValidatorRecord(**sample_validator_record_params).copy(
+            balance=balance,
+            latest_status_change_slot=latest_status_change_slot,
+        )
+        for _ in range(10)
+    ]
+
+    result = min_empty_validator_index(
+        validators=validators,
+        current_slot=current_slot,
+        zero_balance_validator_ttl=zero_balance_validator_ttl,
+    )
+
+    assert result == expected
+
+
+@pytest.mark.parametrize(
+    "expected",
+    (
+        (True),
+        (ValidationError),
+    ),
+)
+def test_validate_proof_of_possession(sample_beacon_state_params, pubkeys, privkeys, expected):
+    state = BeaconState(**sample_beacon_state_params)
+
+    privkey = privkeys[0]
+    pubkey = pubkeys[0]
+    withdrawal_credentials = b'\x34' * 32
+    randao_commitment = b'\x56' * 32
+    domain = get_domain(
+        state.fork_data,
+        state.slot,
+        SignatureDomain.DOMAIN_DEPOSIT,
+    )
+
+    deposit_input = make_deposit_input(
+        pubkey=pubkey,
+        withdrawal_credentials=withdrawal_credentials,
+        randao_commitment=randao_commitment,
+    )
+    if expected is True:
+        proof_of_possession = sign_proof_of_possession(deposit_input, privkey, domain)
+
+        assert validate_proof_of_possession(
+            state=state,
+            pubkey=pubkey,
+            proof_of_possession=proof_of_possession,
+            withdrawal_credentials=withdrawal_credentials,
+            randao_commitment=randao_commitment,
+        )
+    else:
+        proof_of_possession = b'\x11' * 32
+        with pytest.raises(ValidationError):
+            validate_proof_of_possession(
+                state=state,
+                pubkey=pubkey,
+                proof_of_possession=proof_of_possession,
+                withdrawal_credentials=withdrawal_credentials,
+                randao_commitment=randao_commitment,
+            )
+
+
+def test_process_deposit(sample_beacon_state_params,
+                         zero_balance_validator_ttl,
+                         privkeys,
+                         pubkeys):
+    state = BeaconState(**sample_beacon_state_params).copy(
+        slot=zero_balance_validator_ttl + 1,
+        validator_registry=(),
+    )
+
+    privkey_1 = privkeys[0]
```

Comment:
maybe should do a loop

```python
for index in range(3):
    privkey = privkeys[index]
    pubkey = pubkeys[index]
    ....
    result_state, validator_index = process_deposit()
    ...
    check_results()
```

Reply:
Oh I see! Maybe update comment 

```
# Add the third validator.
# Should overwrite previously exited validator
```

or something. Would be good to make it more explicit than just the `assert len(result_state.validator_registry) == 2`